[RFC] tests: migrate legacy test 11

lucc opened the pull request:
There is something strange going on in this test.  At some points I had to add "empty vim commands" in order to stop the test from stalling.  And at one point even that did not help.  Does anyone have an idea about that?

lucc committed:
test: Migrate legacy test 11.

Some parts of the test depend on gzip(1).  They are skipped if gzip is not
available.

Some `:write` and `:edit` commands produce messages and "hit enter" prompts
that had to be treated with an extra `feed('<C-L>')`.  In the original test
file this was not neccessary because it was `:source!`ed.

watiko commented:
@lucc Maybe the same issue (except for `FilterRead{Pre,Post}`) happened to us. (#3877)
To avoid the issue I used `helpers.feed()`.

```lua
-- helpers.execute('command`)
helpers.feed(':command<CR><C-l>')
```

And `FilterRead{Pre,Post}` has another issue. I replaced some `execute()` with one `source()` then I got the error `Vim(!):E493: Backwards range given: 23,$!cat`.

I hope this is helpful for us :)

lucc commented:
@watiko thanks for these comments. I think I fixed it.

lucc commented:
ref: part of #1286

justinmk commented:
@lucc Shouldn't `testdir/test11.*` be deleted (and `testdir/Makefile` updated)?

justinmk commented on test/functional/legacy/011_spec.lua:
```diff
@@ -0,0 +1,231 @@
+local wait = helpers.wait -- only for debugging
+
+local function has_gzip()
+  return os.execute('gzip --help >/dev/null 2>&1') == 0
```
`helpers.eval('executable("gzip")') == 1`

justinmk commented on test/functional/legacy/011_spec.lua:
```diff
@@ -0,0 +1,231 @@
+  teardown(function()
+    os.remove('Xtestfile.gz')
+    os.remove('Xtest.c')
+--    os.remove('foo.c')
```
is this needed?

justinmk commented on test/functional/legacy/011_spec.lua:
```diff
@@ -0,0 +1,231 @@
+      expect('\n'..text1)
+    end)
+
+    it('BufRead{Pre,Post} (using gzip)', function()
```
if it fits on one line, writing it out like `BufReadPre,BufReadPost` makes grepping easier.

justinmk commented:
LGTM after comments and squash.

lucc committed:
test: Remove old files for legacy test 11.

lucc commented:
I also moved the zipping of Xtestfile into the respective test block (and out of setup()) and moved one more test block into the `if gzip then test else pending end` block.

justinmk commented:
Validated. Thanks!
